src/main/java/org/jstepanovic/server/service/Server.java

```diff
@@ -6,50 +6,64 @@
 import java.net.InetAddress;
 import java.net.ServerSocket;
 import java.net.Socket;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.Future;
+import java.net.SocketException;
+import java.util.concurrent.*;
 
 import static org.jstepanovic.commons.Constants.SERVER_ADDRESS;
 import static org.jstepanovic.commons.Constants.SERVER_PORT;
 
 
-public class Server {
+public enum Server {
+    INSTANCE;
 
-    private final int numberOfProcessors;
-
-    private static Server instance;
+    private final int numberOfProcessors = Runtime.getRuntime().availableProcessors();
```

Comment:
just small comment here: usually you want to use max(availableProcessors - 1, 1) for long-running servers to leave some space for garbage collector and other system threads

Reply:
Done